homeassistant/components/homematicip_cloud/climate.py

```diff
@@ -176,4 +187,3 @@ def _get_first_heating_thermostat(heating_group: AsyncHeatingGroup):
     for device in heating_group.devices:
         if isinstance(device, (AsyncHeatingThermostat, AsyncHeatingThermostatCompact)):
             return device
-    return None
```

Comment:
We should return something explicitly in all cases if we return explicitly in any case.

Reply:
Agree, but this code part is unreachable.
The heating_group only exists because there is a thermostat  in devices.
In case there is no wall thermostat in this group there must be a heating thermostat inside.
I have another climate related PR open. 
@MartinHjelmare, if you could review #27772 i will add it back with fixes after review.